src/wp-includes/user.php

```diff
@@ -682,14 +682,25 @@ function count_many_users_posts( $users, $post_type = 'post', $public_only = fal
 		return $pre;
 	}
 
-	$userlist = implode( ',', array_map( 'absint', $users ) );
-	$where    = get_posts_by_author_sql( $post_type, true, null, $public_only );
+	// Cleanup the users array. Remove duplicates and sort for consistent ordering.
+	$users = array_unique( array_filter( array_map( 'absint', (array) $users ) ) );
```

Comment:
We know that `$users` is an `array` here due to the `is_array()` check. So this can be removed.

```suggestion
	$users = array_unique( array_filter( array_map( 'absint', $users ) ) );
```